### PR TITLE
Whitelist certain eventType and scope selector for analytics in scope

### PR DESCRIPTION
**Pull request by zhouyx**
Alternative solution to #8329

As we agreed would only allow `visible` and `custom` trigger at this point. And the selector has to be the analytics parentElement itself. 
 
@dvoytenko @lannka PTAL

### CODE REVIEW

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 1]
```diff
@@ -205,14 +213,31 @@ export class AmpAnalytics extends AMP.BaseElement {
               'attributes are required for data to be collected.');
           continue;
         }
+        // Check for not supported trigger for scoped analytics
+        // TODO(will replace with this.isScoped_ after #8360)
+        const isScoped = true;
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/amp-analytics.js]: DO NOT SUBMIT

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 2]
```diff
@@ -49,6 +50,13 @@ installVariableService(AMP.win);
 
 const MAX_REPLACES = 16; // The maximum number of entries in a extraUrlParamsReplaceMap
 
+const BLACKLIST_EVENT_IN_SCOPE = [
+  AnalyticsEventType.CLICK,
+  AnalyticsEventType.TIMER,
+  AnalyticsEventType.SCROLL,
+  AnalyticsEventType.HIDDEN,
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/amp-analytics.js]: `hidden` should be ok

Comment (zhouyx) [inline extensions/amp-analytics/0.1/amp-analytics.js]: removed

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 1 above)
Comment (zhouyx) [inline extensions/amp-analytics/0.1/amp-analytics.js]: removed blocker

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 3]
```diff
@@ -50,6 +51,12 @@ installVariableService(AMP.win);
 
 const MAX_REPLACES = 16; // The maximum number of entries in a extraUrlParamsReplaceMap
 
+const BLACKLIST_EVENT_IN_SCOPE = [
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Let's call it "blacklist in sandbox" then?

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 4]
```diff
@@ -50,6 +51,12 @@ installVariableService(AMP.win);
 
 const MAX_REPLACES = 16; // The maximum number of entries in a extraUrlParamsReplaceMap
 
+const BLACKLIST_EVENT_IN_SANDBOX = [
```
Comment (lannka) [inline extensions/amp-analytics/0.1/amp-analytics.js]: shall we do whitelist instead?

Comment (zhouyx) [inline extensions/amp-analytics/0.1/amp-analytics.js]: Agree whitelist is easier to maintain. changed.

**Pull request merged**